pkg/mapreconciler/reconciler.go

```diff
@@ -0,0 +1,276 @@
+		if deleted {
+			err := m.m.Delete(entry.Key())
+			// If we error, retry this key later, don't hold up the delete tracker.
+			if err != nil && !errors.Is(err, ebpf.ErrKeyNotExist) {
+				m.retryQueue.AddRateLimited(entry.Key())
+				return nil
+			}
+
+			// clear the key from the retry queue (if present) since we just performed an delete without error
+			m.retryQueue.Forget(entry.Key())
+
+			return nil
+		}
+
+		// If a custom map entry equal function is provided, use it to determine if the entry should be updated.
+		if m.options.mapEntryEqual != nil {
+			v, err := m.m.Lookup(entry.Key())
+			if err == nil {
+				if m.options.mapEntryEqual(entry, entry.Key(), v) {
+					return nil
+				}
+			}
+		}
+
+		err := m.m.Put(entry.Key(), entry.Value())
+		// If we error, retry this key later, don't hold up the delete tracker.
+		if err != nil {
+			m.retryQueue.AddRateLimited(entry.Key())
+			return nil
+		}
+
+		// clear the key from the retry queue (if present) since we just performed an update without error
+		m.retryQueue.Forget(entry.Key())
+
+		return nil
+	})
+	if err != nil {
+		panic(fmt.Errorf("delete tracker bug: %w", err))
+	}
+
+	return rev, invalidate
+}
+
+func (m *mapReconciler[E, K, V]) fullReconciliation(ctx context.Context) uint64 {
+	rxn := m.db.ReadTxn()
+	rev := m.tbl.Revision(rxn)
+	iter, _ := m.tbl.All(rxn)
+	_ = statedb.ProcessEach(iter, func(entry E, _ uint64) error {
+		// If a custom map entry equal function is provided, use it to determine if the entry should be updated.
+		if m.options.mapEntryEqual != nil {
+			v, err := m.m.Lookup(entry.Key())
+			if err == nil {
+				if m.options.mapEntryEqual(entry, entry.Key(), v) {
+					return nil
+				}
+			}
+		}
+
+		err := m.m.Put(entry.Key(), entry.Value())
+		// If we error, retry this key later, don't hold up the delete tracker.
+		if err != nil {
+			m.retryQueue.AddRateLimited(entry.Key())
+			return nil
+		}
+
+		// clear the key from the retry queue (if present) since we just performed an update without error
+		m.retryQueue.Forget(entry.Key())
+
+		return nil
+	})
+
+	// Delete any keys in the map that are not in the stateDB table.
+	mIter := m.m.Iterate()
+	var (
+		v V
+		k K
+	)
+	// we are not allowed to delete while iterating, so we collect the keys to delete and delete
+	// them after the iteration
+	var toDelete []K
+	for mIter.Next(&k, &v) {
+		_, _, found := m.tbl.First(rxn, m.pkIndex.Query(k))
+		if !found {
+			toDelete = append(toDelete, k)
+		}
+	}
+	for _, k := range toDelete {
+		err := m.m.Delete(k)
+		// If we error, retry this key later, don't hold up the delete tracker.
+		if err != nil && !errors.Is(err, ebpf.ErrKeyNotExist) {
+			m.retryQueue.AddRateLimited(k)
+			continue
+		}
+
+		// clear the key from the retry queue (if present) since we just performed an update without error
+		m.retryQueue.Forget(k)
+	}
+
+	return rev
+}
+
+func (m *mapReconciler[E, K, V]) Retirer(ctx context.Context) error {
```

Comment:
Should be `Retrier`? Is there a reason this is exported? 
BTW, would `retryLoop` and `reconcileLoop` work better as names?